### DISCUSSION TITLE
Link Prediction in Narrative Fiction using PyNeuraLogic

### DISCUSSION THREAD

**Discussion by FuzzyNZ** [LukasZahradnik/PyNeuraLogic#31] [Ideas]
A project using works of fiction from the public domain and PyNeuraLogic to predict Character links across texts using relational categories, and integrating entity data generated in Stanford's Stanza NLP Python API. 

Using a combination of Bash loops and Grep we've filtered about 330 keywords through the repository of texts and have used the Stanford Stanza Python library to create dictionaries of entities.

Now that we have a solid data set the goal is to use PyNeuraLogic to predict new links between characters across the texts, using labels such as "Lawyer" or "Detective" or "Thief" etc, and to also use other graphs that include familial, household, and work links.  The idea is to create a kind of mega-narrative from the repository. 

A future roadmap could also include working with a graph of graphs.

**Comment by GustikS** [comment]
Hi, can you please put some context to this discussion/project, e.g. copy out the description from the emails we've exchanged, so that it's self-contained here? We can then continue the discussion altogether here. Thanks!

**Comment by FuzzyNZ** [comment]
Hi Gustav,

Sure thing!  Been flat out finishing a contract this week.  Have free time
next week so looking forward to giving this some quality time.

Best regards,
Steve
